source/playbooks/SC/ssmdocs/SC_S3.1.ts

```diff
@@ -20,6 +20,19 @@ class ConfigureS3PublicAccessBlockDocument extends ControlRunbookDocument {
     });
   }
 
+  /** @override */
+  protected getParseInputStepOutputs(): Output[] {
```

Comment:
I saw this function was rewritten a few times, is there any way to pull it out to reduce duplicate code?

Reply:
These are all overrides of the base class definition from [here](https://github.com/aws-solutions/aws-security-hub-automated-response-and-remediation/blob/develop/source/playbooks/SC/ssmdocs/control_runbook.ts). The derived classes only define how they differ. I'm not terribly happy with the abstraction since you need to understand really well what the "default control runbook" does and how it works in order to extend it. I wouldn't want to introduce more inheritance to the mix without a better plan, but the different possible outputs of the parse input step could probably be sourced from a single helper rather than defined in every override.